Libraries/libbom/Sources/bom.c

```diff
@@ -268,7 +285,9 @@ bom_index_add(struct bom_context *context, const void *data, size_t data_len)
 
     /* Update length for newly added index. */
     index_header->count = htonl(ntohl(index_header->count) + 1);
-    header->index_length = htonl(ntohl(header->index_length) + sizeof(struct bom_index));
+    if (resize) {
+      header->index_length = htonl(ntohl(header->index_length) + sizeof(struct bom_index));
```

Comment:
Because if we are not resizing, we are using free space within the existing allocated size, which is index_length.

If we have run out of pre-allocated slots and are resizing, index_length is increasing by `sizeof(struct bom_index)`